src/test/functional/steps/court-types.ts

```diff
@@ -0,0 +1,66 @@
+import {Then, When} from 'cucumber';
+import * as I from '../utlis/puppeteer.util';
+import {expect} from 'chai';
+
+
+When('I click the types tab', async () => {
+  const selector = '#tab_court-types';
+  const elementExist = await I.checkElement(selector);
+  expect(elementExist).equal(true);
+  await I.click(selector);
+});
+
+Then('I can view the existing court types', async () => {
+  const elementExist = await I.checkElement('#courtTypesForm');
+  expect(elementExist).equal(true);
+
+  const tabClosed = await I.checkElement('#court-types.govuk-tabs__panel--hidden');
+  expect(tabClosed).equal(false);
+});
+
+
+When ('I check a court type', async() =>{
+  const selector = '#court_types-3';
+  const elementExist = await I.checkElement('#court_types-3');
+  expect(elementExist).equal(true);
+
+  await I.click(selector);
+
+});
+
+Then('I click on save', async () => {
+  const selector = 'button[name="saveCourtTypes"]';
+  const elementExist = await I.checkElement(selector);
+  expect(elementExist).equal(true);
+  await I.click(selector);
+});
+
+Then('a green update message is displayed showing Court Types updated', async () => {
+  const elementExist = await I.checkElement('#courtTypesTab .govuk-panel--confirmation');
+  expect(elementExist).equal(true);
+});
+
+When ('I uncheck a court type', async() =>{
+  const selector = '#court_types-3';
+  const elementExist = await I.checkElement('#court_types-3');
+  expect(elementExist).equal(true);
+
+  await I.click(selector);
```

Comment:
Should we check here that the court type is indeed unchecked?

Reply:
Create a new helper method isElementChecked in puppeteer.utils.ts n and altered tests accordingly